File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int n = cost.size();
        int first = cost[0];
        int second = cost[1];
        int temp = 0;
        for(int i=2; i<n; i++){
            temp = second;
            second = cost[i] + min(first, second);
            first = temp;
        }

        return min(first, second);
    }
};

File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.go
func minCostClimbingStairs(cost []int) int {
    n := len(cost)
    steps := make([]int, n+1)
    
    for i, num := range cost {
        steps[i] = num
    }

    for i := 2; i <= n; i++ {
        steps[i] += min(steps[i-1], steps[i-2])
    }

    return steps[n]
}